docs/design/2021-04-21-unify-log-library.md

```diff
@@ -0,0 +1,234 @@
+# Proposal: Unifying Log Library
+
+- Author(s): [Yifan Xu](https://github.com/SabaPing)
+- Last updated:  Apr 12, 2021
+- Discussion at: https://github.com/pingcap/tidb/pull/24181
+- Tracking issue: https://github.com/pingcap/tidb/issues/24190
+
+## Motivation or Background
+
+Except for slow query logs, all other logs must satisfy the [unified-log-format RFC standard](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md).
+
+However, in practice, it was found that the format of logs is confusing, as shown in the following four points:
+
+- There are few logging configuration instructions in the document. We need to enrich document especially the type and the format of logs each component would emit.
+- The configured logging parameters do not match the runtime logging, e.g. `tidb_stderr` is configured with text format, but the log is in json format.
+- The logs of some components do not meet the [unified-log-format RFC standard](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md), e.g. `tiflash_cluster_manager`.
+- Duplicate logs, e.g. `pd_stderr` will emit both text and json logs with duplicate content (but with a few subtle differences in timestamps).
+
+## Logging code for each component
+
+### [pingcap/log](https://github.com/pingcap/log)
+
+As a common logging library for PingCAP golang projects, it does the following things:
+
+- Provides the standard config schema.
+- Provides a factory method for creating log handlers.
+- Hard code the log format according to [unified-log-format RFC standard](https://github.com/tikv/rfcs/blob/master/text/2018-12-19-unified-log-format.md).
```

Comment:
```suggestion
- Hard code the log format according to [unified-log-format RFC standard](https://github.com/tikv/rfcs/blob/23d4f9aed68a295b678e8bd909ee8479e3ba0bd1/text/2018-12-19-unified-log-format.md).
```

use a fixed commit hash to prevent affected by master changes